serialization.rst

```diff
@@ -222,6 +225,70 @@ possible mapping values are:
   if it exists, will be sent as an associative array to the ``bsonUnserialize``
   function to initialise the object's properties.
 
+Scalar Types
+------------
+
+For types that the driver has a wrapper class for, it is possible to define a
+map from each supported data type to a class of your own, providing it
+implements the ``MongoDB\BSON\TypeWrapper`` interface.
+
+The supported data types are Binary, Decimal128, Javascript, MaxKey, MinKey,
+ObjectID, Regex, Timestamp, and UTCDateTime.
+
+The ``MongoDB\BSON\TypeWrapper`` interface defines two functions:
+``createFromBSONType()``, a factory method which takes a ``MongoDB\BSON\Type``
+argument, and ``toBSONType()``.
+
+As an example, a wrapped UTCDateTime class, could look like::
+
+    class UTCDateTimeWrapper implements \MongoDB\BSON\TypeWrapper
+    {
+        private $intern;
+
+        public function __construct( \MongoDB\BSON\UTCDateTime $type )
+        {
+            $this->intern = $type->toDateTime();
+        }
+
+        static function createFromBSONType(\MongoDB\BSON\Type $type)
+        {
+            if (! $type instanceof \MongoDB\BSON\UTCDateTime) {
+                throw new UnexpectedValueException;
+            }
+
+            return new UTCDateTimeWrapper( $type );
+        }
+
+        function toBSONType()
+        {
+            return new \MongoDB\BSON\UTCDateTime( $this->intern );
+        }
+    }
+
+
+If the defined class wraps (composes) an original ``MongoDB\BSON\*`` class,
+then it SHOULD also implement the accompanying
+``\MongoDB\BSON\<classname>Interface`` interface.
+
+The type interfaces include all the type-specific methods from the
+original class, with the exact same arguments and return types. For example,
+they will not include the ``__construct()`` and ``__debugInfo()`` methods.
+
+For example, a user-defined ``UTCDateTimeWrapper`` class needs to implement the
+``MongoDB\BSON\UTCDateTimeInterface`` and ``MongoDB\BSON\TypeWrapper``
+interfaces.
+
+It is not necessary to return an object from the ``createFromBSONType()``
+"factory" method. You may return anything that can be serialized to BSON,
+including an object that implements ``MongoDB\BSON\Serializable``, but you MAY NOT
+return an object that implements ``MongoDB\BSON\TypeWrapper``. Returning a
+different type from the original one, could be used as a way to migrate
```

Comment:
"original one, could" doesn't seem like a grammatically correct use of a comma. I'd just remove it entirely.